src/components/Quote.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+
+const Quote = () => {
+  const [quote, setQuote] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    // Define the API URL with the "funny" category
+    const apiUrl = 'https://api.api-ninjas.com/v1/quotes?category=funny';
+
+    // Fetch the quote from the API
+    fetch(apiUrl)
+      .then((response) => response.json())
+      .then((data) => {
+        // Check if the response data is empty
+        if (data.length === 0) {
+          setQuote('No funny quote available');
+        } else {
+          // Extract the quote from the response data
+          const randomQuote = data[0]?.quote || 'I think there is something in the fact that it is hard to be good looking and funny. You have to have an oddball quality people have to sympathise with you to find you funny.';
+          setQuote(randomQuote);
```

Comment:
- [ ] [Optional][Highly recommended] The API request of quotes always returns a `400` error `bad request` because the API you used requires an `API key` to retrieve data from it.

here is the solution:
```javascript
fetch(apiUrl, {
      headers: {
        'X-Api-Key': 'your API key', // you will have this key from your account settings (see the image below)
      },
    })
```

![image](https://github.com/gilberthappi/math-magicians/assets/68796232/471f623d-0cd2-41ef-a08b-998781d5941f)